packages/data-context/src/sources/FileDataSource.ts

```diff
@@ -32,12 +36,14 @@ export class FileDataSource {
 
   async getFilesByGlob (cwd: string, glob: string | string[], globOptions?: GlobbyOptions) {
     const globs = ([] as string[]).concat(glob).map((globPattern) => {
+      const workingDirectoryPrefix = path.join(cwd, path.sep)
+
       // If the pattern includes the working directory, we strip it from the pattern.
       // The working directory path may include characters that conflict with glob
       // syntax (brackets, parentheses, etc.) and cause our searches to inadvertently fail.
       // We scope our search to the working directory using the `cwd` globby option.
-      if (globPattern.startsWith(cwd)) {
-        return globPattern.replace(cwd, '.')
+      if (globPattern.startsWith(workingDirectoryPrefix)) {
+        return globPattern.replace(workingDirectoryPrefix, '')
       }
 
       return globPattern
```

Comment:
Would `return path.relative(workingDirectoryPrefix, globPattern)` also work? Might not even need line 39 at that point

Reply:
I think `path.relative` has the potential to mutate the glob patterns to be relative to the current execution directory, which may or may not equal the given `cwd` argument. In those cases, the pattern would no longer be representative of what the user expects.